Framework/Algorithms/src/DiscusMultipleScatteringCorrection.cpp

```diff
@@ -534,17 +534,17 @@ double DiscusMultipleScatteringCorrection::interpolateSquareRoot(const Histogram
 /**
  * Interpolate function using flat interpolation from previous point
  */
-double DiscusMultipleScatteringCorrection::interpolateFlat(const Mantid::HistogramData::Histogram &histToInterpolate,
-                                                           double x) {
-  if (x > histToInterpolate.x().back()) {
-    return histToInterpolate.y().back();
+double DiscusMultipleScatteringCorrection::interpolateFlat(
+    std::shared_ptr<const Mantid::HistogramData::Histogram> histToInterpolate, double x) {
+  if (x > histToInterpolate->x().back()) {
```

Comment:
There's lots of derefs through all of this. Can we simply take a ref at the start to cut the noise and number of derefs the compiler has to do down?
```
auto &xHisto = histToInterpolate->x()
auto &yHisto = histToInterpolate->y()

if (x < xHisto.back())
// ...etc.
```